tests/util/kartutil.go

```diff
@@ -0,0 +1,273 @@
+		return false, nil
+	}
+}
+
+// K8sDaemonSetCheck Check if the daemonset exists and is ready
+func K8sDaemonSetCheck(dsname string, ns string, timeout int) error {
+	return waitForCondition(timeout, isDaemonSetReady(dsname, ns))
+}
+
+// K8sDeploymentCheck Check if the deployment exists and is ready
+func K8sDeploymentCheck(depname string, ns string, timeout int) error {
+	return waitForCondition(timeout, isDeploymentReady(depname, ns))
+}
+
+// K8sGetPods Check if Pods exists and is/are Running
+func K8sGetPods(podstr string, ns string, timeout int) ([]string, error) {
+	pods := []string{}
+	for t := 0; t <= timeout; t++ {
+		podList, err := k8sClient.K8sClientset.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			log.Errorf("k8s list pods failed. error=%s", err)
+			return nil, err
+		}
+		pods = []string{}
+		for _, p := range podList.Items {
+			if p.Status.Phase != v1.PodRunning {
+				continue
+			}
+			if strings.HasPrefix(p.ObjectMeta.Name, podstr) {
+				pods = append(pods, p.ObjectMeta.Name)
+			} else if match, _ := regexp.MatchString(podstr, p.ObjectMeta.Name); match {
+				pods = append(pods, p.ObjectMeta.Name)
+			}
+		}
+		if timeout == 0 || len(pods) > 0 {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if len(pods) == 0 {
+		return nil, errors.New("pod not found")
+	}
+	return pods, nil
+}
+
+// K8sExecInPod Exec into the pod. Output: stdout, stderr, err
+func K8sExecInPod(pod string, ns string, cmd []string) (string, string, error) {
+	req := k8sClient.K8sClientset.CoreV1().RESTClient().Post().Resource("pods").Name(pod).Namespace(ns).SubResource("exec")
+	option := &v1.PodExecOptions{
+		Command: cmd,
+		Stdout:  true,
+		Stderr:  true,
+		TTY:     true,
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(k8sClient.Config, "POST", req.URL())
+	if err != nil {
+		return "", "", err
+	}
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdout: buf,
+		Stderr: errBuf,
+	})
+	return buf.String(), errBuf.String(), nil
+}
+
+func init() {
+	if !isK8sEnv() {
+		log.Error("could not find k8s env")
+	}
+}
+
+/*
+func StartKubearmor(k8sMode bool) error {
+	if !k8sMode {
+		log.Errorf("only k8s mode supported")
+		return errors.New("only k8s mode supported")
+	}
+	err := K8sDaemonSetCheck("kubearmor", "kube-system", 20)
+	if err == nil {
+		log.Info("kubearmor daemon already executing, skipping install")
+		return nil
+	}
+	log.Println("installing kubearmor")
+	err = k8sInstallKubearmor()
+	if err != nil {
+		log.Errorf("start kubearmor failed error=%s", err)
+		return err
+	}
+	err = K8sDaemonSetCheck("kubearmor", "kube-system", 20)
+	if err == nil {
+		return nil
+	}
+	return nil
+}
+*/
+
+// Kubectl execute
+func Kubectl(cmdstr string) (string, error) {
+	cmdf := strings.Fields(cmdstr)
+	cmd := exec.Command("kubectl", cmdf...)
+	sout, err := cmd.Output()
+	return string(sout), err
+}
```

Comment:
Just wanted to ensure for some reason if the client-go APIs are not ready then kubectl exec is available and the tester should not add exec themselves. currently apply/delete are kubectl execed because client-go code to handle them is complex and hence pending.